### PR TITLE
Profile page fixes

### PR DESCRIPTION
**Pull request by GCheung55**
Fix/verify #51 - [Profile page] Favorite/unfavorite article
* Favoriting/unfavoriting worked but caused bug where the whole list of articles is replaced by the one article that was favorited/unfavorited.
* Fixed ^ issue by replacing how articles are loaded via `store.query` to loading from the relationships. The issue was that `loadArticles` made a request via `store.query` and then setting the result as `articles`. The `profile` model maintains a relationship to the `articles`, so when an article is pushed into the store with a relationship to the model, the array set by `loadArticles` is replaced completely. Relying on the relationship to load articles prevents this bug from appearing.
* Updated the styling of the favorite article button to be consistent with spec.

Verifies #52 - [Profile page] Follow/unfollow profile
* There's an acceptance test for following/unfollowing via profile, as a logged in user and as an anonymous user.

Replaced the `favorite-article` component with a `favorite-article-button` component.
* Replaced to follow DDAU, reducing the responsibilities of the component.
* Updated styles because the button looks different in feed-item vs in an article.

Updated profile template to access correct properties.

Updated profile controller follow/unfollow tasks to leverage profile model's follow/unfollow methods, thus reducing code.

-----

@Alonski I apologize for a large PR - the more changes I made the more things got tangled up.

### CODE REVIEW

File: app/templates/components/favorite-article-button.hbs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<button class="btn btn-sm {{@class}}" {{action @onClick}} data-test-favorite-article-button>
```
Comment (Alonski) [inline app/templates/components/favorite-article-button.hbs]: Two things :)
First is that we shouldn't be using `@prop` as this is an Octane feature. Right? @mansona Any ideas?
Second I prefer not to override the name of something like `class` with `@class`. Usually in my apps I name it to something like `wrappingClass`.
What do you think?

Comment (GCheung55) [reply app/templates/components/favorite-article-button.hbs]: According to [Ember 3.1 and 3.2 Beta Released](https://blog.emberjs.com/2018/04/13/ember-3-1-released.html) blog post, named arguments, aka `@prop`, was introduced in Ember@3.1.

Can you help me understand how `class` is being overridden? `class` is a property set on the `favorite-article-button` and it's being treated as a property that is being passed in - thus the usage of `{{@class}` to identify it as such.

Comment (GCheung55) [reply app/templates/components/favorite-article-button.hbs]: As we discussed, I will remove usages of named args and add back in tagNames.

The reason as I understand it is that named args was made for Octane and empty tagNames string usage is not in the guides.

**Comment by Alonski** [comment]
Thanks @GCheung55 !

**Pull request merged**